src/Client.php

```diff
@@ -264,23 +264,30 @@ public function getIdealIssuers(): IdealIssuers
      * Only completed orders could be refunded.
      *
      * @param string $order_id
+     * @param Properties\Amount $anount
      * @return array
      * @throws \GingerPluginSdk\Exceptions\InvalidOrderStatusException
      * @throws \GingerPluginSdk\Exceptions\RefundFailedException
      */
-    public function refundOrder(string $order_id)
+    public function refundOrder(string $order_id, Properties\Amount $amount = null)
     {
         $order = $this->getOrder(id: $order_id);
 
         if ($order->getStatus()->get() != "completed") {
             throw new InvalidOrderStatusException($order->getStatus()->get(), 'completed');
         }
 
-        if (!$order->getCurrentTransaction()->isCaptured()) {
+        if ($order->getCurrentTransaction()->isCapturable() && !$order->getCurrentTransaction()->isCaptured()) {
```

Comment:
I think this condition better to divide by two cases, first one is when transaction is not captured, and next one is when transaction can't be captured. By this way we can return more valuable exception

Reply:
"By this way we can return more valuable exception"
I don't think so. With two divided checks we will have exception when order doesn't require any capturing. 
To my mind this should work only for capturable orders